pkg/cmd/provisioning/utils.go

```diff
@@ -173,9 +175,22 @@ func GetListOfCredentialsRequests(dir string) ([]*credreqv1.CredentialsRequest,
 			// infrastructure.
 			if value, ok := cr.Annotations["release.openshift.io/delete"]; ok && value == "true" {
 				log.Printf("Ignoring CredentialsRequest %s/%s as it is marked for in-cluster deletion", cr.Namespace, cr.Name)
-			} else {
-				credRequests = append(credRequests, cr)
+				continue
 			}
+
+			// Handle CredentialsRequest with the feature-gate annotation
+			if value, ok := cr.Annotations[featureGateAnnotation]; ok {
+				if !enableTechPreview {
+					log.Printf("Ignoring CredentialsRequest %s/%s with tech-preview annotation", cr.Namespace, cr.Name)
+					continue
+				}
+				if value != string(configv1.TechPreviewNoUpgrade) {
+					log.Printf("Ignoring CredentialsRequest %s/%s with tech-preview value %s", cr.Namespace, cr.Name, value)
+					continue
+				} // else allow it to be added it to the list of CredReqs to process
+			}
```

Comment:
IIUC, first we should check if `featureGateAnnotation` value is equal to `TechPreviewNoUpgrade` and then check if tech preview is enabled?

```suggestion
			// Handle CredentialsRequest with the feature-gate annotation
			if value, ok := cr.Annotations[featureGateAnnotation]; ok {
				if value != string(configv1.TechPreviewNoUpgrade) {
					log.Printf("Ignoring CredentialsRequest %s/%s with release.openshift.io/feature-gate annotation value %s", cr.Namespace, cr.Name, value)
					continue
				}
				if !enableTechPreview {
					log.Printf("Ignoring CredentialsRequest %s/%s with release.openshift.io/feature-gate annotation value %s", cr.Namespace, cr.Name, configv1.TechPreviewNoUpgrade)
					continue
				}
				// else allow it to be added to the list of CredentialsRequests to process
			}
```